obci/interfaces/bci/analysis_master.py

```diff
@@ -0,0 +1,222 @@
+            chunk: numpy 2D data array (channels × samples)
+            target: name of the target
+        """
+        classifier.learn(chunk, target)
+
+    # ----------------
+    # internal methods
+    # ----------------
+
+    @log_crash
+    def __init__(self, addresses):
+        super(AnalysisMaster, self).__init__(addresses=addresses, type=peers.ANALYSIS)
+        # initialize parameters of the subclass
+        self.init_params()
+        channel_count = len(self.config.get_param('channel_names').split(';'))
+        self.buffer = self.create_buffer(channel_count, self._buffer_ret_func)
+
+        # load classifier from file, if requested
+        self.wisdom_path = self.config.get_param('wisdom_path') if self.config.has_param('wisdom_path') else None
+        self._classifier = self._get_classifier()
+
+        # array of requests to classification thread
+        self.queue_classification = Queue()
+        # array of requests to learning thread
+        self.queue_learning = Queue()
+
+        # separate thread for classification
+        classification = Thread(target=self._run_classification)
+        classification.daemon = True
+        classification.start()
+
+        # separate thread for learning
+        learning = Thread(target=self._run_learning)
+        learning.daemon = True
+        learning.start()
+
+        self.ready()
+
+    def _buffer_ret_func(self, blink, chunk=None):
+        """Internal function for the buffer object.
+
+        Args:
+            blink: information contained in a single blink
+            chunk: numpy 2D data array (channels × samples)
+        """
+        if chunk is None:
+            chunk = blink
+            blink = None
+        request = {'chunk': chunk, 'blink': blink}
+        target = self.identify_blink(blink)
+        if target is None:
+            self.queue_classification.put(request)
+        else:
+            request['target'] = target
+            self.queue_learning.put(request)
+
+    def _get_classifier(self):
+        """Provide a classifier object from file or create a new one.
+
+        Returns:
+            AbstractClassifier. a classifier object
+        """
+        if self.config.has_param('wisdom_load') and os.path.isfile(self.wisdom_path):
+            return pickle.load(self.wisdom_path)
+        else:
+            return self.create_classifier()
+
+    def _run_classification(self):
+        """This method will be run as a separate thread."""
+        while True:
+            request = self.queue_classification.get()
+            probabilities = self.classify(self._classifier, request['chunk'], request['blink'])
+            if probabilities is not None:
+                self.add_result(request['blink'], probabilities)
+
+    def _run_learning(self):
+        """This method will be run as a separate thread."""
+        while True:
+            request = self.queue_learning.get()
+            clone = self._classifier.clone()
```

Comment:
Yes, and the logic you're writing about can be implemented by overriding AnalysisMaster's _learn_ method.